src/Microsoft.Framework.ConfigurationModel/Sources/CommandLineConfigurationSource.cs

```diff
@@ -33,9 +34,9 @@ public override void Load()
             string key, value;
             var argIndex = 0;
 
-            while (argIndex < Args.Length)
+            while (argIndex < Args.Count())
```

Comment:
Hmm any reason not to use a `foreach`? There are a ton of calls to `.ElementAt()`, which is kind of a gross method to call.